### ISSUE TITLE
[Card] Add link text

### ISSUE THREAD

**Issue by shixiedesign** [carbon-design-system/ibm-dotcom-library#2196]
A feature request for component [Card](https://ibmdotcom-react.mybluemix.net/?path=/story/patterns-sub-patterns-card--link-clickable): currently only an icon is allowed for bottom left corner of card. We'd like to allow link text for potential future case cases. 

@photodow suggests to **simply implement a styled text and icon instead** of using existing [link with icon](https://ibmdotcom-react.mybluemix.net/?path=/story/components-link-with-icon--default) component. This is because: 

1. the added link text should be 14px from bottom, 16px from left of card (see spec below)
2. keep icon 16x16px from the bottom left corner when there's no link text
3. link text should have NO underline on hover

### Spacing

![image](https://user-images.githubusercontent.com/15144993/80248588-398f2c00-863e-11ea-81c8-e71ce9e30653.png)

### Visual spec

Selected screenshots only. For full visual spec of **Card**, please see Abstract collection: https://share.goabstract.com/f757e306-109b-43d6-a044-7154f87f7a7f

![image](https://user-images.githubusercontent.com/15144993/82916541-283c8680-9f40-11ea-9654-0cef821d5bc8.png)
![image](https://user-images.githubusercontent.com/15144993/82916591-3be7ed00-9f40-11ea-9d0d-65d167bf2934.png)

**Comment by sethibm** [comment]
@shixiedesign @photodow Thanks for sharing this.  It seems like all elements in any pattern should at least be made up of other formalized patterns or components, rather than simply be styled within a given component or pattern?  

We otherwise may wind up with various bits and pieces in given patterns, groups, etc that lose connection to a cohesive system.  I've been trying to map and identify a few of these kind of things the past two days.

One of the current considerations is to set the default padding on lower level components like "Link with icon" to just safe / neutral padding in its standalone state, and then that padding is removed and changed to zero when it sits within another pattern like Cards, which allows that parent pattern to set particular padding like 16px left and 14px below.  

If we did the above, then I think the Link with icon could be used in this situation?  

For removing underline in a particular usage of Link with icon, it seems this could be of universal interest for this component.  @photodow, could this be added as an override / knob within the component?

**Comment by RobertaJHahn** [comment]
Shixie will investigate the current state and update the issue to clearly describe the requirement.

**Comment by shixiedesign** [comment]
Hi @RobertaJHahn I've updated the issue's title and description: 

- this is a feature request to **add Link text as an option to cards**
- suggestion from Seth (above comment) is to use **Link + icon component**, and suggestion from James is to implement this _without_
- I will leave you guys to decide how to implement as long as requirements for visuals and interactions specified above at met

**Comment by sethibm** [comment]
Thanks @shixiedesign.  I'm not aware of reasoning from @photodow on why it's better to implement without using Link + icon component.  Maybe this is a good topic to consider in latest ontology discussion?

Question: Should all elements within any pattern be formally defined as patterns themselves (ie in atomic manner)?  Or will Library continue to have various styled elements, with no formal pattern definition or option to use in repetition within other patterns?  

Three examples of current situation: 1) this link + icon text within card 2) The header and subheader text in ContentBlock 3) The header in ContentGroup.  There are likely others.

I may not understand the advantages of simply styling some parts of patterns, rather than defining, formalizing, and making them re-usable.  

cc @larahanlon2 @wonilsuhibm

**Comment by photodow** [comment]
@sethibm two more reasons I didn't see provided (and these would be more of the actual reasoning I would think) from a code standpoint it's not valid HTML to have an `<a>` as a child of `<a>`, and by using a link with icon you create this invalid HTML. The other issue is the designs by my understanding don't have interactions on the text and icon. The card itself gets the interaction, and the text and icon are static. We could create a static variant to link with an icon if we want, but so far this is the only place we'd be using it that I'm aware of.

**Comment by larahanlon2** [comment]
@sethibm 
From a design POV, James is correct – we want the whole card to be clickable, not the link + icon. And we don't want both to be clickable. For this card, the link + icon acts as an indicator rather than a call-to-action which is what it would do if used independently. 

We discussed if we could use the link + icon component and disable the interactivity for this use case. However, my understanding is that disabling it would inherit the actual disabled state of the component which we don't wait either. Doing that would require an override.

Based on all the variables and the design intent, I agree with the suggestion of having **a styled text and icon** instead of using the existing link + icon component here.

**Comment by sethibm** [comment]
@larahanlon2 @photodow Ok, yeah, it seems implementing key overrides would be needed for a system of patterns made up of patterns.  It otherwise requires pieces of patterns to be styled per use case.

It's not a direct connect, but this example and others seem a bit like Detach from Symbol in Sketch libraries.  It's a red, maybe yellow flag imho...ie workable case to case, maybe not ideal system-wise.

**Comment by RobertaJHahn** [comment]
@larahanlon2 Can you summarize the conclusion for Dev.  Then we can move this issue forward.  Thanks.

**Comment by larahanlon2** [comment]
@RobertaJHahn + dev team. 

# Summary 

### As per Shixie's original FR:
Currently, only an icon is allowed for bottom left corner of card. We'd like to allow link text for potential future case cases.

@photodow suggests to simply implement **a styled text and icon instead of using existing link with icon component**. This is because:

- the added link text should be 14px from bottom, 16px from left of card (see spec below)
- keep icon 16x16px from the bottom left corner when there's no link text
- link text should have NO underline on hover

### Additional info:
- We discussed if we could use the link + icon component and disable the interactivity for this use case. However, my understanding is that disabling it would inherit the actual disabled state of the component which we don't want.
- We also don't want to override the component behavior.
- The whole card should be clickable, not the link + icon.
